docs/corpus.markdown

```diff
@@ -1,45 +1,166 @@
+|  year|  title|  part| path                       |
+|-----:|------:|-----:|----------------------------|
+|  1997|     26|     1| data/clean/97-26-01.txt    |
+|  1997|     26|     2| data/clean/97-26-02.txt    |
+|  1997|     26|     3| data/other/Y1997T26P02.txt |
 
-The second important feature of the driver is that it provides a function named `stream` The `stream` function should return an [iterable](https://docs.python.org/3/glossary.html#term-iterable)—in most cases, a [generator](https://docs.python.org/3/glossary.html#term-generator)—that emits the index value (or values) and text of each document in the corpus. Thus, the first item emitted by `driver.stream()` in the CFR corpus might be `(1975, 1, 1), "Text of the 1975 CFR, Title 1, Part 1"`.
+Note that no particular pattern is needed for the file path; they may be as
+esoteric as your data. Note also that the use of absolute paths in an index csv
+will likely hamper portability.
 
-Drivers may implement other features (such as only streaming a subset of documents based on the index), but these will be non-standard, and estimators should not expect them as a matter of course.
+The assuming the file above is saved in `data/index.csv`, the `driver.py` file
+for this corpus would be as follows:
 
-## Corpus Metadata
+``` {.python}
+import quantgov as qg
 
-Relevant metadata will vary from corpus to corpus. Metadata can be generated from one of two sources: from the text itself, or from additional external information. In the first case, the best practice is to write scripts that understand the corpus driver interface, and can therefore be used in other corpora. An example of this approach can be seen in the `get_wordcount.py` and `get_restriction_count.py` in the QuantGov generic corpus and most corpora based on it. In the second case, the external resources should be stored in a [databank](http://docs.quantgov.org/databank) kept separate from the corpus itself, which the corpus scripts treat as read-only. An example of this approach is the agency attribution in the CFR corpus, which relies on a set of documents separate from the main CFR text.
+from pathlib import Path
 
-## Writing a new Corpus
+BASE_DIR = Path(__file__).resolve().parent
 
-The easiest way to write a new corpus is to fork the most similar official corpus (see below) and modify it to represent the relevant body of text. There are three principle problems to solve in the creation of a new corpus:
+driver = qg.corpus.IndexDriver(index=BASE_DIR.joinpath('data', 'index.csv'))
+```
 
-1.  How can the text be obtained and, if necessary, translated to plain text?
-2.  What is the logical unit of analysis for the corpus?
-3.  How can the text be organized to most usefully reflect the unit of analysis.
+### Custom Drivers
```

Comment:
I was pretty lost with this paragraph. Maybe we need an example, or just cut it if it is something that we think people will never use.